### PR TITLE
IT BELONGS IN A MUSEUM: Adds Museum Trophy Cases, and Soapstone Messages

### PR DESCRIPTION
**Pull request by Iamgoofball**
# librarian before goofcode

librarian2

# librarian after goofcode

rs_1024x759-140326100842-1024 2harrison-ford-indiana-jones

:cl: Iamgoofball
add: Added Chisels to the Librarian! Use it on floors or walls to leave messages for future crewmembers they can read with the examine verb.
add: Added Museum Trophy Cases to the library! Hit these with items, and type in a plaque, to turn these into trophies! Once it goes in, it stays forever in the pool. You can put an item back into the pool and off a display case by replacing it with a new item.
/:cl:

:cl: Kor
add: The Librarian now has access to the treasure hunter's hat/uniform/coat, as well as the curator's whip.
/:cl:

dreamseeker_2016-12-11_19-41-30

dreamseeker_2016-12-11_19-39-41

### CODE REVIEW

**Comment by KorPhaeron** [comment]
I'm going to merge this when it's up to standard the idea is great I'm going to ignore all comments about "ruining the theme" of the least popular job

File: code/modules/library/lib_items.dm
Comment (ChangelingRain) [inline code/modules/library/lib_items.dm]: a chisel, pah! true messages would use a soapstone!

File: code/modules/library/lib_items.dm
Comment (ChangelingRain) [inline code/modules/library/lib_items.dm]: return ..()

File: code/modules/library/lib_items.dm
Comment (Iamgoofball) [inline code/modules/library/lib_items.dm]: k

File: code/modules/library/lib_items.dm
Comment (ChangelingRain) [inline code/modules/library/lib_items.dm]: also you'll need something to make sure chisel messages don't spawn on turfs that are.... not floors or walls

**Comment by RandomMarine** [comment]
Should be able to unanchor and pull them around to try and desperately save the displays from a loose singulo or something.

**Comment by francinum** [comment]
does this sanitize mind-containing items?

**Comment by GunHog** [comment]
Someone has been playing Dark Souls...

**Comment by Iamgoofball** [comment]
@francinum Variables aren't kept
@RandomMarine They're indestructible just like your virginity

**Comment by ChangelingRain** [comment]
Chisel messages also need to record what station they're on, so if at all possible they're at least sort of relevant instead of "these hallways are long as all fuck" on ministation, from dream

**Comment by RandomMarine** [comment]
If they're somehow singulo-proof then 👎

**Comment by KorPhaeron** [comment]
Reminder to 

-Limit placing items to the librarian somehow (ID lock would be okay, but still easily bypassed)

-A limit to the number of items you can add each round to the pool to prevent it being flooded

-Rename the Librarian something like Curator or Archivist

File: code/game/objects/structures/displaycase.dm
Comment (optimumtact) [inline code/game/objects/structures/displaycase.dm]: this is a sleep, check the item is not qdeleted and or already has an item inside it after this.

File: code/modules/library/lib_items.dm
Comment (optimumtact) [inline code/modules/library/lib_items.dm]: IT'S A SLEEP BOB

File: code/modules/library/lib_items.dm
Comment (optimumtact) [inline code/modules/library/lib_items.dm]: it's a SLEEP BOB

**Comment by ExcessiveUseOfCobblestone** [comment]
@optimumtact can you explain what a sleep is and how to "fix" the sleep?

**Comment by GunHog** [comment]
They should NOT be indestructible! When the case is destroyed, simply say that the item is old and crumbled to dust or something; Delete the item when the case is broken.

This is to create the situation where a greyshirt can break in and smash all the cases to start a fight with the librarian!

**Comment by Shadowlight213** [comment]
Is there a limit or something to this? Also how often can something be added per round?
I don't want some memer to add everything on the station to it to the point where loading the list lags the server.

**Comment by ExcessiveUseOfCobblestone** [comment]
are the messages suppose to be a DS ref

**Comment by PKPenguin321** [comment]
librarian should actually get a whip

**Comment by KorPhaeron** [comment]
@PKPenguin321 

Patience

File: code/game/objects/structures/displaycase.dm
Comment (Iamgoofball) [inline code/game/objects/structures/displaycase.dm]: aight

**Comment by Iamgoofball** [comment]
@ExcessiveUseOfCobblestone Great Chest Ahead!

**Comment by ExcessiveUseOfCobblestone** [comment]
meme

**Comment by optimumtact** [comment]
@ExcessiveUseOfCobblestone when I say sleep I actually mean that the method your calling can cause the current proc to block and stop executing, allowing other game logic to run in the meantime.

What it means practically is that when your code in that method starts to execute again the state of the underlying object may have changed (from some other event, i.e the object got blown up)

So you need to redo any sanity checks for state you have, as well as check the object is not qdeleted.

**Comment by dannno** [comment]
cool idea

**Comment by AnturK** [comment]
I'm ambivalent about this, librarian needs some love but persistence features are awful.

**Comment by Iamgoofball** [comment]
@anturk It's all 100% cosmetic and flavor stuff.

**Comment by AnturK** [comment]
Oh you can't take them out ? Sounds good then.

**Comment by Iamgoofball** [comment]
Yup, once it goes in it never comes out.

**Comment by dannno** [comment]
I assume there's a blacklist for things like the disk?

**Comment by Iamgoofball** [comment]
Yup

**Comment by PKPenguin321** [comment]
maybe hold off on the disarm whip
or make it just not disarm
or make it only have a chance to disarm and slightly less damage
or make it do very little damage and disarm

**Comment by Iamgoofball** [comment]
I literally give zero fucks about balance meme and will be ignoring it completely.

**Comment by jughu** [comment]
I cant wait to see shitcueity chiseled in the brigs everywhere

**Comment by HankMoodyExperience** [comment]
>not being able to construct display cases;
>cant move them around
>indestructible

m u h i n d e s t r u c t i b l e e x p l o i t s
make the cases appear roundstart for the librarian, as constructible, destroyable things, of like wood and glass or something.

and limit chisel use so some memer doesnt literally plaster an entire station with "nigger" carvings

**Comment by KorPhaeron** [comment]
Display case blocks a vent on meta

There needs to be some way to remove items from the pool both in case someone fills the entire thing with 1k variants of wgw and to stop the pool growing forever. Maybe have the case destructible and lose the item inside if its blown up

Still need a limit on items added per round

**Comment by Iamgoofball** [comment]
You can't add the same item over and over again in one round
I'll make cases explode
You can smash messages to erase them from the pool, I'll lower their health
Limiting items added per round kind of sucks but I guess I can look into it

**Comment by KorPhaeron** [comment]
Merge my PR

**Review by YakumoChen — COMMENTED** [review]
>Librarian gets free security armor and a guaranteed disarm now
>Implying that powergaming tiders won't use this every shift for the free armor

**Comment by KorPhaeron** [comment]
Just like people who play CMO every round for the free space suit and infinite stun right?

**Comment by RandomMarine** [comment]
Suggesting a rework.

Make each display start out empty to begin with. When an item is first added, the case remains unlocked and the item easily retrievable. No blacklist necessary.

If the item remains inside the display case when the round ends, the item's preserved inside and the case is locked the following rounds, impossible to retrieve what's inside.

Each display has a keypad with a code shared between them that's randomized each round. Librarian spawns with a paper with that code. When the code's punched in, the item in the display is destroyed and it's unlocked, allowing something new to be placed inside.

Finally, no big pool of items that will get bloated with WGW papers or junk, what's put in each individual display stays until it's destroyed.

**Comment by YakumoChen** [comment]
CMO is a head of staff, though. Librarian is just another assistant with a fancier name.

**Comment by Iamgoofball** [comment]
@YakumoChen I don't care, find somewhere else to whine.

**Comment by Iamgoofball** [comment]
@RandomMarine I'll think about it.

**Comment by optimumtact** [comment]
Great job kor, you ruined the PR

File: code/game/machinery/vending.dm
Comment (coiax) [inline code/game/machinery/vending.dm]: STEALTH CHANGES!?!?!?

**Comment by coiax** [comment]
- I think you should make chisel messages in Box Station only appear on Box Station and so on, otherwise it loses all meaning.
- You should add (real world) dates and times of when the trophies were added and the character name, along with storing (internally) the ckey of who added it, for subsequent justice if they added the Bee Movie script and/or Nazis.
- The trophy pool will get very large, I think you should maybe limit to only one new trophy a round for sanity's sake. (So, assuming a librarian always trophies and 1 hour rounds, that's 168+ trophies a week). Bonus points if librarians can vote up trophies that they approve of, (COMMEND ME).
- Admins will need some method of viewing the trophy pool and PURGING AND PUNISHING.

Otherwise, :+1:x:100:, this is exactly the sort of fun shit that is cool.

**Comment by FantasticFwoosh** [comment]
**_TAKE ALL MY MONEY._**

Though i sort of wish the curator role wasn't so heavily attributed to the librarian/archeologist and more a thing anybody could take up. **Using the the chisel for Goof-Archeology when?**

> this pr comments are starting to get full so put additional non-coding feedback here please on this forum thread https://tgstation13.org/phpBB/viewtopic.php?f=10&t=8849

File: code/modules/library/lib_items.dm
Comment (AnturK) [inline code/modules/library/lib_items.dm]: Give this a proper name instead of forced reference that makes no sense in context.

**Comment by Iamgoofball** [comment]
>I think you should make chisel messages in Box Station only appear on Box Station and so on, otherwise it loses all meaning.

Already done

File: code/modules/library/lib_items.dm
Comment (AnturK) [inline code/modules/library/lib_items.dm]: target instead of !target

File: code/controllers/subsystem/persistence.dm
Comment (AnturK) [inline code/controllers/subsystem/persistence.dm]: No cap on this ? Station will be one giant scribble.

File: code/game/objects/structures/displaycase.dm
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm]: forceMove

File: code/game/objects/structures/displaycase.dm
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm]: cent_bar ?

File: code/modules/clothing/glasses/glasses.dm
Comment (AnturK) [inline code/modules/clothing/glasses/glasses.dm]: ???

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/clothing/glasses/glasses.dm
Comment (Iamgoofball) [inline code/modules/clothing/glasses/glasses.dm]: KORRRRRRRRR

File: code/modules/library/lib_items.dm
Comment (Iamgoofball) [inline code/modules/library/lib_items.dm]: Fuck

File: code/modules/library/lib_items.dm
Comment (Iamgoofball) [inline code/modules/library/lib_items.dm]: I was told by a maintainer to make it a soapstone

File: code/game/objects/structures/displaycase.dm
Comment (Iamgoofball) [inline code/game/objects/structures/displaycase.dm]: Wanted to make sure that you couldn't open it

**Comment by Haevacht** [comment]
As long as it's still a chisel and not the soapstone.

**Comment by Iamgoofball** [comment]
It's staying a soapstone it's literally a dark souls joke

File: code/modules/clothing/glasses/glasses.dm
Comment (KorPhaeron) [inline code/modules/clothing/glasses/glasses.dm]: Wasnt me

Comment (KorPhaeron) [inline code/modules/clothing/glasses/glasses.dm]: https://github.com/tgstation/tgstation/pull/22109/files

It IS in there, not sure why it shows as a difference

**Comment by Haevacht** [comment]
why

**Comment by karlnp** [comment]
Are they singulo-proof?

**Comment by Jazaen** [comment]
You should add some way of removing items from the pool, in case some asshole just labels a bunch of floor tiles idiotic stuff and throws them into the case to make sure nothing interesting ever appears.

File: code/modules/library/lib_items.dm
Comment (Iamgoofball) [inline code/modules/library/lib_items.dm]: I will

File: code/game/objects/structures/displaycase.dm
Comment (Shadowlight213) [inline code/game/objects/structures/displaycase.dm]: Should double check that the centcom official/ert doesn't have this.
Actually, what would even happen if you removed the item? Would it not go back in the pool?

Comment (Iamgoofball) [inline code/game/objects/structures/displaycase.dm]: It would let you get items from other rounds

**Comment by ExcessiveUseOfCobblestone** [comment]
I want people to go explore space now so they can put erotic messages below the Venus statues.

**Comment by Dorsisdwarf** [comment]
I love this except for the librarian getting armor for no reason.

Its a crumpled archaeology hat not combat gear you dork he doesnt need armor if he whips the gun out of the syndicate thief's hand

**Comment by dannno** [comment]
it does seem pretty stupid for it to have armor for no actual reason
it's just a hat and some clothes

**Comment by Shadowlight213** [comment]
Don't forget the force 15 weapon that makes the target drop all held items on hit!
The officer cannot arrest if you disable his hand!

**Comment by KorPhaeron** [comment]
My thought process behind giving him basic armour was that he'd have a little protection if he ran around in lavaland or something looking for artifacts

**Comment by KorPhaeron** [comment]
>complaining about force 15 when there is a public lathe on meta that produces infinite force 15 weapons

The disarm part is the good part, and it's still worse than a baton.

**Comment by dannno** [comment]
the whip doesn't seem like a problem at all to me
it would be better if it was a 1 tile reach weapon with disarm that did less damage if adjacent to the target, but that's not a thing (yet)

**Comment by Shadowlight213** [comment]
The weapon is already really good. Useful on crew more than lavaland though.
The librarian actually going to lavaland is unlikely unless you add mining access to him.
The armor isn't needed.

**Comment by Iamgoofball** [comment]
~~literally don't care~~

**Comment by Dorsisdwarf** [comment]
An archaeologist who needs armor to rescue totems belongs in a museum

File: code/modules/library/lib_items.dm
Comment (optimumtact) [inline code/modules/library/lib_items.dm]: please fix the path name as well

File: code/game/objects/structures/displaycase.dm
Comment (Iamgoofball) [inline code/game/objects/structures/displaycase.dm]: ale mao

File: code/modules/clothing/glasses/glasses.dm
Comment (optimumtact) [inline code/modules/clothing/glasses/glasses.dm]: did you squash merge kor's pr?

**Comment by jughu** [comment]
@Iamgoofball whip better be a ranged weapon for 2 tiles

**Comment by TehZombehz** [comment]
What, nobody made the obvious 'try tongue but hole' dark souls soapstone reference? Can't wait till this is testmerged and ~~get to play archeologist as librarian~~  ~~roll librarian to find all the tools were simply stolen by quite literally anybody and find the actual librarian job isn't needed for any of this~~ have fun.

File: code/modules/clothing/glasses/glasses.dm
Comment (Iamgoofball) [inline code/modules/clothing/glasses/glasses.dm]: yes

Comment (optimumtact) [inline code/modules/clothing/glasses/glasses.dm]: Probably why, especially if it had a master merge

Comment (Iamgoofball) [inline code/modules/clothing/glasses/glasses.dm]: Ale Mao

File: code/game/objects/items/weapons/melee/misc.dm
Comment (kevinz000) [inline code/game/objects/items/weapons/melee/misc.dm]: Seems VERY powerful.

**Comment by Armhulen** [comment]
goofball PLEASE finish this, just one last push until it's done

**Comment by optimumtact** [comment]
Extract soapstoan/chisel into it's own pr

extract display cases into it's own pr.

use json for the persistence.

Thanks

**Comment by Iamgoofball** [comment]
@optimumtact 
no
no
if i can get it working I will

**Comment by optimumtact** [comment]
If you don't seperate them you don't get a merged pr.

This isn't a request, it's an order

**Comment by Iamgoofball** [comment]
They're both for the librarian only though

**Pull request closed without merging**
